File: Day4/part2/solution_part2.go
package main

import (
    "bufio"
    "fmt"
    "log"
	"os"
	"strings"
	"sort"
)

func main() {
    file, err := os.Open("../input.txt")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

	counter := 0
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
		line := strings.Fields(scanner.Text())
		m := make(map[string]int)
		for _, element := range line {
			e := SortString(element)
			m[e] = 0
		}
		if len(line) == len(m) {
			counter += 1
		}
	}
	
	fmt.Println(counter)

    if err := scanner.Err(); err != nil {
        log.Fatal(err)
    }
}

func SortString(w string) string {
    s := strings.Split(w, "")
    sort.Strings(s)
    return strings.Join(s, "")
}

File: Day5/solution_part2.py
with open('input.txt') as f:
    content = [int(x) for x in f.readlines()]

counter = 0
index = 0
while index < len(content):
    offset = content[index]
    if content[index] >= 3:
        content[index] -= 1
    else:
        content[index] += 1
    index += offset
    counter += 1

print counter